.storybook/main.js

```diff
@@ -1,15 +1,11 @@
-
 module.exports = {
-  "stories": [
-    "../stories/**/*.stories.js"
-  ],
-  "addons": [
+  core: { builder: "@storybook/builder-vite" },
+  stories: ["../stories/**/*.stories.jsx"],
```

Comment:
Vite requires stories to have the `jsx` extension.

> Be sure that any files containing JSX syntax use a .jsx or .tsx file extension, which [vite requires](https://vitejs.dev/guide/features.html#jsx). This includes .storybook/preview.jsx if it contains JSX syntax.

See https://github.com/storybookjs/builder-vite#migration-from-webpack--cra